### PR TITLE
DR-1942 Update retrieveDataset endpoint to include azure metadata

### PR DESCRIPTION
**Pull request by myessail**

### CODE REVIEW

File: src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java [hunk 1]
```diff
@@ -24,11 +40,22 @@
...
+  private static final String AZURE_DATASET_ID = "<storageAccount>.<dataset>";
+
   private static final String DEPLOYED_APPLICATION_RESOURCE_ID =
       "/subscriptions/<subscription>/resourceGroups"
           + "/<resource_group>/providers/Microsoft.Solutions/applications/<application_name>";
 
-  private MetadataDataAccessUtils() {}
+  private static ResourceService resourceService;
+
+  private static AzureBlobStorePdao azureBlobStorePdao;
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java]: In general, ~you shouldn't have~ it's rare to have non-final `static` fields. In this case these are also injected objects that have state, so it doesn't make sense to use `static` fields for them.

File: src/main/java/bio/terra/service/resourcemanagement/azure/AzureStorageAccountService.java [hunk 2]
```diff
@@ -339,4 +340,17 @@ void deleteCloudStorageAccount(
             storageAccountResource.getApplicationResource().getAzureResourceGroupName(),
             storageAccountResource.getName());
   }
+
+  public AzureStorageAccountResource getStorageAccount(
```
Comment (samanehsan) [inline src/main/java/bio/terra/service/resourcemanagement/azure/AzureStorageAccountService.java]: It looks like this isn't getting called anywhere. Can we remove it?

File: src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java [hunk 3]
```diff
@@ -43,21 +70,87 @@ private MetadataDataAccessUtils() {}
...
+    AccessInfoModel accessInfoModel = new AccessInfoModel();
+
+    BlobContainerClientFactory targetDataClientFactory =
+        azureBlobStorePdao.getTargetDataClientFactory(
+            profileModel, storageAccountResource, ContainerType.METADATA, false);
+
+    // Given the sas token, rebuild a signed url
+    BlobSasTokenOptions options =
+        new BlobSasTokenOptions(
+            Duration.ofMinutes(15),
```
Comment (tlangs) [inline src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java]: Should this time be configurable from the environment, or are we ok with it being 15 minutes and not going to change it?

Comment (nmalfroy) [reply src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java]: Ultimately, we should have this be a dataset or snapshot or request option but we should have a followon story to make Sas token times congurable

File: src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java [hunk 4]
```diff
@@ -43,21 +70,83 @@ private MetadataDataAccessUtils() {}
...
+            new BlobSasPermission().setReadPermission(true).setListPermission(true),
+            AzureSynapsePdao.class.getName());
+    String signedURL =
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java:110]: I think we have a common signed URL method we should use here (it does the added thing of setting content disposition with the user name for tracing)

File: src/main/resources/api/data-repository-openapi.yaml [hunk 5]
```diff
@@ -3506,6 +3506,8 @@ components:
       properties:
         bigQuery:
           $ref: '#/components/schemas/AccessInfoBigQueryModel'
+        azure:
```
Comment (nmalfroy) [inline src/main/resources/api/data-repository-openapi.yaml]: I think this should probably be `parquet`

**Review by nmalfroy — CHANGES_REQUESTED** [review]
Nice!  main change think we want is the openapi spec

File: src/test/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+  @Mock private BlobSasUrlFactory blobSasUrlFactory;
+
+  private Dataset azureDataset;
+  private DatasetSummary azureDatasetSummary;
+  private BillingProfileModel defaultProfileModel;
+
+  @Before
+  public void setup() throws Exception {
+    MockitoAnnotations.openMocks(this).close();
```
Comment (nmalfroy) [inline src/test/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtilsTest.java]: Not sure what this line does?  I think you could do away with this by adding the:
```@RunWith(MockitoJUnitRunner.class)``` annotation to the class

File: src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java [hunk 7]
```diff
@@ -43,21 +68,98 @@ private MetadataDataAccessUtils() {}
...
+                    .map(
+                        t -> {
+                          String tableBlob =
+                              new ST(AZURE_BLOB_TEMPLATE).add("table", t.getName()).render();
+                          String unsignedTableUrl =
+                              new ST(AZURE_PARQUET_LINK)
+                                  .add("storageAccount", storageAccountResource.getName())
+                                  .add("blob", tableBlob)
+                                  .render();
+                          //                          String tableUrl =
```
Comment (samanehsan) [inline src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java]: Can we take out this comment?

File: src/test/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+            eq(ContainerType.METADATA),
+            any(),
+            any()))
+        .thenReturn(
+            "https://michaelstorage.blob.core.windows.net/metadata/parquet/sample/signedUrl");
+
+    AccessInfoParquetModel infoModel =
+        metadataDataAccessUtils.accessInfoFromDataset(azureDataset).getParquet();
+
+    Assert.assertEquals("test-dataset", infoModel.getDatasetName());
```
Comment (samanehsan) [inline src/test/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtilsTest.java:100]: You could replace these repeated values with variables, but it's a pretty small test so I'll leave that up to you!

**Review by samanehsan — APPROVED** [review]
Lgtm! ☔

**Review by nmalfroy — APPROVED** [review]
okeydoke, looks good to me!

**Pull request merged**